packages/components/src/components/teaser-media/teaser-media.ts

```diff
@@ -0,0 +1,165 @@
+    super.updated(changedProperties);
+  }
+
+  onHover() {
+    if (this.hasSlotController.test('expandable') && this.shadowRoot) {
+      const hiddenDiv = this.shadowRoot.getElementById('expandable')!;
+      hiddenDiv.setAttribute(
+        'style',
+        'height: auto; visibility: visible; opacity: 1; margin-top: 16px; margin-bottom: 16px;'
+      );
+    }
+  }
+
+  onHoverEnd() {
+    if (this.shadowRoot) {
+      const hiddenDiv = this.shadowRoot.getElementById('expandable')!;
+      hiddenDiv.setAttribute('style', 'height: 0; visibility: invisible; opacity: 0; margin-top: 0; margin-bottom: 0;');
+    }
+  }
+
+  render() {
+    const slots = {
+      'teaser-has-default': this.hasSlotController.test('[default]'),
+      'teaser-has-media': this.hasSlotController.test('media'),
+      'teaser-has-meta': this.hasSlotController.test('meta'),
+      'teaser-has-headline': this.hasSlotController.test('headline'),
+      'teaser-has-expandable': this.hasSlotController.test('expandable')
+    };
+
+    return html`
+      <div
+        class="relative flex flex-col max-w-[862px] overflow-hidden"
+        part="base"
+        @mouseenter=${this.onHover}
+        @mouseleave=${this.onHoverEnd}
+      >
+        <div class=${cx('mb-4', !slots['teaser-has-media'] && 'hidden')} part="media">
+          <slot name="media"></slot>
+        </div>
+
+        <div class="absolute flex flex-col justify-end h-full w-full pb-4">
+          <div
+            class=${cx(
+              'flex-1 opacity-[80%]',
+              this.variant === 'gradient-white' && 'bg-gradient-to-t from-white/75 to-55%',
+              this.variant === 'gradient-dark' && 'bg-gradient-to-t from-primary-800/75 to-55%'
+            )}
+          ></div>
+          <div
+            class=${cx(
+              'opacity-[80%]',
+              {
+                white: 'bg-white ',
+                'neutral-100': 'bg-neutral-100',
+                primary: 'bg-primary text-white',
+                'primary-100': 'bg-primary-100',
+                'gradient-white': 'bg-white/75',
+                'gradient-dark': 'bg-primary-800/75 text-white'
+              }[this.variant]
+            )}
+          >
+            <div class=${cx('flex-col text-left p-4')} part="content">
+              <div part="meta" class=${cx('gap-2 mb-4', !slots['teaser-has-meta'] && 'hidden')}>
+                <slot name="meta"></slot>
+              </div>
+
+              <div part="headline" class=${cx('text-lg font-bold m-0')}>
+                <slot name="headline"
+                  >Always insert one semantically correct heading element here (e. g. &lt;h2&gt;)</slot
+                >
+              </div>
+
+              <div
+                id="expandable"
+                class=${cx(
+                  slots['teaser-has-default'] &&
+                    'h-[0px] invisible opacity-0 hidden md:[transition:_height_0.2s_linear,opacity_0.1s_linear_0.1s] md:block',
```

Comment:
just realized, that in the last sample, the expandable content is not visible after leaving the hover state even though the purple marked slots in the docs is still visible for a few ms...

nevertheless I feel like the transition should be aligned.

Reply:
@karlbaumhauer in which browser did you view it? The timing of the slots/parts wasn't perfect for me either, but I assume it's because of how the component is structured in the storybook. If you look at the correct examples at the end, I think the transitions are correct?

Reply:
@saemik94 I use Microsodt Edge and yes, the sample is fine. Nevertheless, the documentation seems buggy can you investigate a bit more, why this happens? I feel like having "broken" docs is also not a nice thing to have in our library.

If you dont find the issue or it happens because some technical issues, pls add a comment to the documentation in the beginning to let users know, that we will need to investigate/fix it later and they should look at the sample. Pls then also add it to the known broser issues (in case its browser related).